### PR TITLE
H-938: Add app-wide error boundary with fallback message

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

We have an error boundary around blocks, showing an error message with 'reset' button if a block's code throws an error. But we didn't have one any higher than that in the app tree, meaning that any other crashes would tank the app in production, showing NextJS's default 'A client-side error has occured' screen.

This PR adds an error boundary around the page component with a fallback component, so that the rest of the UI is maintained and the user has a reset button and can see the error id (for identification in Sentry) and error message – see video in demo.

As a drive-by I have also fixed a console error we were gettin for "Invalid Sentry DSN: undefined".

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

I have freestyled the appearance of this component and it could look a lot better.

## 🛡 What tests cover this?

- None.

## ❓ How to test this?

1. Throw an error somewhere in the code.

## 📹 Demo

Here I throw an error when the mention suggester is used. Note that the modal which appears first only happens in development – in production, only the error fallback component will be shown.

https://github.com/hashintel/hash/assets/37743469/b152a2c0-50a1-446b-9865-c5cba3b200e6

### CODE REVIEW

**Review by nonparibus — DISMISSED** [review]
Functional LGTM; thanks!

File: apps/hash-frontend/src/pages/_app.page/error-fallback.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        We've been notified and will investigate — please click the button below
+        to reset the page to its initial state.
+      </Typography>
+      <Box sx={{ textAlign: "center" }}>
+        <Button onClick={resetError} size="small">
+          Reset the page
+        </Button>
+      </Box>
+      <Typography mt={2}>
+        You can also <Link href="/contact">contact us</Link>. If you do, please
```
Comment (benwerner01) [inline apps/hash-frontend/src/pages/_app.page/error-fallback.tsx:76]: We don't have a `/contact` page, is this in anticipation of having moved from `app.hash.ai` to `hash.ai`?

**Review by benwerner01 — APPROVED** [review]
Thanks Ciaran, looks good and is working for me locally 👍

File: apps/hash-frontend/src/pages/_app.page/error-fallback.tsx (hunk 1 above)
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/_app.page/error-fallback.tsx]: Oops, nope – I've updated the link in https://github.com/hashintel/hash/pull/3368/commits/88dfc66eab15fa87567a80b1d1c0abff46db8396, thank you

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**